### PR TITLE
Replace time with std::time in `components/devtools`

### LINKED ISSUE #30150: Replace uses of `time` (`precise_time_ns`) with `std::time`, `chrono`, or the new `time` crate
We have a lot of uses of a very old version of the `time` crate. We should gradually replace all of these uses with calls to `std::time` or if that isn't sufficient with `chrono` or the updated version of the `time` crate. We already depend on those crates.

Some crates should not be modified such as:

 - `components/style`
 - `components/selectors`
 - `third_party/webrender`

These are vendored from Gecko and we need to make the fixes upstream.

### PR DESCRIPTION
**Pull request by Taym95**
---

- [X] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #30150 

- [x] These changes do not require tests because they do not change behavior.

### CODE REVIEW

File: components/devtools/Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ name = "devtools"
 path = "lib.rs"
 
 [dependencies]
+chrono = "0.4"
```
Comment (mrobinson) [inline components/devtools/Cargo.toml:14]: Is it possible to use `std::time` instead of introducing a new dependency in this crate on the chrono crate?

Comment (Taym95) [reply components/devtools/Cargo.toml:14]: I tried, the only missing function is `to_rfc3339 `, can not found it in time and it exists in chrono

Comment (mrobinson) [reply components/devtools/Cargo.toml:14]: Okay. Would it be possible to use `chrono` only for this call to `to_rfc3339`? That way we limit our use of third-party libraries to the minimum and can replace or standardize in the future more easily.

Comment (mrobinson) [reply components/devtools/Cargo.toml:14]: I would only keep `chrono` for `components/devtools/actors/network_event.rs` which is where you need `to_rfc3339` and use `std::time` structs in all other places. I would use something like `Instant` all over the place and only convert to a `chrono` type right before you need the string version of the timestamp.

Comment (Taym95) [reply components/devtools/Cargo.toml:14]: Thanks for suggestion, I used it.

File: components/devtools/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,7 @@ name = "devtools"
 path = "lib.rs"
 
 [dependencies]
```
Comment (mrobinson) [inline components/devtools/Cargo.toml:13]: Please remove `time` from the dependencies as well. It looks like there is one more use in `network_event.rs` before that's possible.

File: components/devtools/actors/network_event.rs [hunk 3]
```diff
@@ -370,7 +371,18 @@ impl NetworkEventActor {
...
+                "{}",
+                (Local::now() -
+                    Duration::from_secs(
+                        self.request
+                            .startedDateTime
+                            .duration_since(UNIX_EPOCH)
+                            .unwrap_or_default()
+                            .as_secs()
+                    ))
+                .to_rfc3339()
```
Comment (mrobinson) [inline components/devtools/actors/network_event.rs:384]: This doesn't seem like the right way to calculate the local time. You are creating a Local from the start time. You are creating a local time here and subtracting the timestamp of the start time. I think that you should look into `Local::timestamp_opt` in order to create the `Local`. In addition, I think that millisecond accuracy would be a better fit here.

**Review by mrobinson — CHANGES_REQUESTED** [review]
This is looking good. I have just a few issues with the current patch.

File: components/devtools/actors/network_event.rs (hunk 3 above)
Comment (Taym95) [reply components/devtools/actors/network_event.rs]: Indeed that was not right, thanks for proposing `timestamp_opt`, I used it.

File: components/devtools/Cargo.toml (hunk 2 above)
Comment (Taym95) [reply components/devtools/Cargo.toml:13]: removed it

File: components/devtools/actors/network_event.rs [hunk 4]
```diff
@@ -366,11 +370,24 @@ impl NetworkEventActor {
...
     pub fn event_actor(&self) -> EventActor {
         // TODO: Send the correct values for startedDateTime, isXHR, private
+        let duration = self
+            .request
+            .startedDateTime
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default();
+
+        let started_datetime_rfc3339 =
+            match Local.timestamp_opt(duration.as_secs() as i64, duration.as_nanos() as u32) {
```
Comment (mrobinson) [inline components/devtools/actors/network_event.rs:380]: I think you are now including the duration twice when calculating the local time. Nanoseconds should only include the portion of the time that isn't represented by the seconds part.

Comment (Taym95) [reply components/devtools/actors/network_event.rs:380]: I see, I can use `timestamp_millis_opt` and provide `millis`, as you mentioned that millisecond accuracy would be a better fit here."

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Thanks! This looks good now.

**Pull request merged**